hibernate5/src/main/java/com/fasterxml/jackson/datatype/hibernate5/HibernateProxySerializer.java

```diff
@@ -182,8 +191,28 @@ protected Object findProxied(HibernateProxy proxy)
                 if (_mapping != null) {
                     idName = _mapping.getIdentifierPropertyName(init.getEntityName());
                 } else {
-                    final SessionImplementor session = init.getSession();
-                    if (session != null) {
+                    // Detect abi compatibility if not set yet
+                    if(hibernateAbiIncompatible == null){
+                        try {
+                            init.getSession();
+                            hibernateAbiIncompatible = false;
+                        } catch (NoSuchMethodError e) {
+                            hibernateAbiIncompatible = true;
+                        }
+                    }
+                    final Object sessionObj;
+                    if(!hibernateAbiIncompatible){
+                        sessionObj = init.getSession();
+                    }else{
+                        try {
+                            Method method = init.getClass().getMethod("getSession");
```

Comment:
Similarly here it should be possible do this just once, locating method is not free (reflection API is somewhat inefficient wrt lookups); should be doable along with earlier detection.